Building on Arch Linux fails with PlatformNotSupportedException

When trying to track down the commit causing https://github.com/ppy/osu-framework/issues/2189, I found that building the framework with Arch Linux fails with `System.PlatformNotSupportedException: arch-x64`.

Git-bisect gives https://github.com/ppy/osu-framework/commit/799563c2b370dc06753be86bd5323b9ace897180:
```bash
799563c2b370dc06753be86bd5323b9ace897180 is the first bad commit
commit 799563c2b370dc06753be86bd5323b9ace897180
Author: miterosan <miterosan@gmail.com>
Date:   Sun Dec 2 14:14:18 2018 +0100

    Add support for sourcelink 3

:040000 040000 6add647e21b3d17d632f50390a197cdb6b6f397f 88c8ed466d4094e9c31f5aa34035d16aad7ccf8d M	osu.Framework
```

Full build output:
```bash
[mkroening osu-framework]$ git clean -xdff && LD_LIBRARY_PATH="$(pwd)/osu.Framework/bin/Debug/netcoreapp2.2" dotnet build osu.Framework
Removing osu.Framework/bin/
Removing osu.Framework/obj/
Microsoft (R) Build Engine version 15.9.20.63311 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for /home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj...
  Generating MSBuild file /home/mkroening/Development/osu-regression/osu-framework/osu.Framework/obj/osu.Framework.csproj.nuget.g.props.
  Generating MSBuild file /home/mkroening/Development/osu-regression/osu-framework/osu.Framework/obj/osu.Framework.csproj.nuget.g.targets.
  Restore completed in 459.02 ms for /home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj.
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly. [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.PlatformNotSupportedException: arch-x64 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RuntimeIdMap.GetNativeLibraryDirectoryName(String runtimeIdentifier) in /_/src/Microsoft.Build.Tasks.Git/RuntimeIdMap.cs:line 26 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.GetNativeLibraryDirectory() in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 56 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 31 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    --- End of inner exception stack trace --- [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_buf_free(GitBuf buf) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Repository.Discover(String startingPath) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RepositoryTasks.LocateRepository(LocateRepository task) in /_/src/Microsoft.Build.Tasks.Git.Operations/RepositoryTasks.cs:line 50 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.LocateRepository.Execute() in /_/src/Microsoft.Build.Tasks.Git/LocateRepository.cs:line 22 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]

Build FAILED.

/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly. [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.PlatformNotSupportedException: arch-x64 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RuntimeIdMap.GetNativeLibraryDirectoryName(String runtimeIdentifier) in /_/src/Microsoft.Build.Tasks.Git/RuntimeIdMap.cs:line 26 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.GetNativeLibraryDirectory() in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 56 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 31 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    --- End of inner exception stack trace --- [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_buf_free(GitBuf buf) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Repository.Discover(String startingPath) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RepositoryTasks.LocateRepository(LocateRepository task) in /_/src/Microsoft.Build.Tasks.Git.Operations/RepositoryTasks.cs:line 50 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.LocateRepository.Execute() in /_/src/Microsoft.Build.Tasks.Git/LocateRepository.cs:line 22 [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
/home/mkroening/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/mkroening/Development/osu-regression/osu-framework/osu.Framework/osu.Framework.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:02.07
```

--- a/osu.Framework/osu.Framework.csproj
+++ b/osu.Framework/osu.Framework.csproj
@@ -33,7 +33,7 @@
   <ItemGroup Label="Package References">
     <PackageReference Include="Markdig" Version="0.15.7" />
     <PackageReference Include="FFmpeg.AutoGen" Version="4.1.0.2" />
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta2-18618-05">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime;build;native;contentfiles;analyzers</IncludeAssets>
     </PackageReference>

